src/pal/tools/gen-buildsys-gcc.sh

```diff
@@ -159,7 +159,7 @@ fi
 overridefile=gcc-compiler-override.txt
 
 # Determine the current script directory
-__currentScriptDir="$(cd -- "$(dirname -- "$0")" && pwd -P)"
+__currentScriptDir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```

Comment:
This is not a correct POSIX script. `BASH_SOURCE` is not available in Bourne shell. Can you show the error you are getting with current script?

Reply:
Same change is in clang script

-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
CMake Error at /usr/share/cmake-3.5/Modules/CMakeCInformation.cmake:94 (include):
  include could not find load file:

    bin/obj/Linux.x64.Debug/gcc-compiler-override.txt
Call Stack (most recent call first):
  CMakeLists.txt:9 (project)

Reply:
Strange, I am unable to get that error in the master branch with gcc or while working on #23111 (I could not have pulled that PR with this error). When did you first encounter this error? Was there any change in your environment?

> Same change is in clang script

`gen-buildsys-gcc.sh` is a `Bash` script (and it is using `BASH_SOURCE[0]` since that line was added), this is a `POSIX` script. They are two different interpreters.

The change you are proposing is a syntax error in POSIX shell: "Bad substitution".

Reply:
it showed up as of this morning. Here is my master branch:

2f2fd8d21f (origin/master, origin/HEAD) Add EntryPointFilter option (#23054)

and the branch where I work:

https://github.com/franksinankaya/coreclr/tree/gcc_compile_rebase

Issue showed up this morning.

Reply:
I just did a rebase. If go to an older commit issue is not there.

commit 7b7f8ec1e3f974a43e96c3f01cb0d5a2da913e17
Author: Jan Vorlicek <janvorli@microsoft.com>
Date:   Thu Mar 7 17:31:39 2019 -0800

    Enable build with clang static analyzer

    As part of LLVM, there is a clang static analyzer tool that provides
    useful analysis of potential issues in the source code base.

    This change enables building coreclr with that analyzer enabled.

This is the commit that made the change on the clang side.

Reply:
Issue was the build.sh change for GCC. Reverting source fixed the issue.